### ISSUE TITLE
Bookmarks - 'Go to page' dialog on empty list

### ISSUE THREAD

**Issue by hius07** [koreader/koreader#7276]
* KOReader version: 2021.01.1-28
* Device: Kindle PW2

#### Issue
'Go to page' dialog should not be called
#### Steps to reproduce
Bookmarks empty window - long tap in the bottom of the screen, on 'No choices available'.

I managed to call the dialog with not empty bookmark list, but it is unstable.

![1](https://user-images.githubusercontent.com/62179190/107733062-b04b0200-6d02-11eb-8db6-f2720a027cb4.png)
---
![2](https://user-images.githubusercontent.com/62179190/107733072-b80aa680-6d02-11eb-8dff-f3a45a185403.png)
---
![3](https://user-images.githubusercontent.com/62179190/107733084-bf31b480-6d02-11eb-8206-258ec8229494.png)

**Comment by hius07** [comment]
The same behaviour on the empty Favorites window.

**Comment by poire-z** [comment]
Solvable by this quick hack (because I don't want to get much into why Button would still have hold_input work when enabled=false):
```diff
--- a/frontend/ui/widget/menu.lua
+++ b/frontend/ui/widget/menu.lua
@@ -1026,2 +1026,3 @@ function Menu:updatePageInfo(select_number)
             self.page_info_text:setText(FFIUtil.template(_("Page %1 of %2"), self.page, self.page_num))
+            self.page_info_text.hold_callback = nil
         else
@@ -1042,2 +1043,3 @@ function Menu:updatePageInfo(select_number)
         self.page_info_text:setText(_("No choices available"))
+        self.page_info_text.hold_callback = function() end -- prevent hold_input
     end
```

**Comment by NiLuJe** [comment]
> (because I don't want to get much into why Button would still have hold_input work when enabled=false):

It *should* be blocking `hold_callback` (or not, depending on `allow_hold_when_disabled`), bit not `hold_input`/`hold_input_func`.

I have no idea what `hold_input`s are used for ;p.

**Comment by poire-z** [comment]
Well, yes, just didn't want to check all instances, if there are cases where any could be enabled=false and if they were using hold_input.
But I guess the logical way of doing it is:
```lua
 function Button:onHoldSelectButton() 
     if not self.enabled and not self.allow_hold_when_disabled) then 
         return true -- or false, dunno
     end
     if self.hold_callback then 
         self.hold_callback() 
     elseif self.hold_input then 
         self:onInput(self.hold_input, true) 
     elseif type(self.hold_input_func) == "function" then 
         self:onInput(self.hold_input_func(), true) 
end 
```

**Comment by NiLuJe** [comment]
At a quick glance, the only users are those Menu/KVP instance with a "click me for pagination" widgets, so that should be pretty-safe.

Because of the weird-ass `readonly` thingy in Button, I'd do it the other way around, but, essentially, yes, agreed ;).

```lua
function Button:onHoldSelectButton() 
	if self.enabled or self.allow_hold_when_disabled then
		if self.hold_callback then 
			self.hold_callback() 
		elseif self.hold_input then 
			self:onInput(self.hold_input, true) 
		elseif type(self.hold_input_func) == "function" then 
			self:onInput(self.hold_input_func(), true) 
		end 
	end
	if self.readonly ~= true then 
		return true 
	end 
end 
```

**Comment by poire-z** [comment]
So, also in onHoldRelease ? (haven't given it any thought :/)

```lua
function Button:onHoldSelectButton()
    if self.enabled or self.allow_hold_when_disabled then
        if self.hold_callback then
            self.hold_callback()
        elseif self.hold_input then
            self:onInput(self.hold_input, true)
        elseif type(self.hold_input_func) == "function" then
            self:onInput(self.hold_input_func(), true)
        end
    end
    if self.readonly ~= true then
        return true
    end
end

function Button:onHoldReleaseSelectButton()
    if self.enabled or self.allow_hold_when_disabled then
        -- Safe-guard for when used inside a MovableContainer,
        -- which would handle HoldRelease and process it like
        -- a Hold if we wouldn't return true here
        if self.hold_callback or self.hold_input or type(self.hold_input_func) == "function" then
            return true
        end
    end
    return false
end
```

**Comment by NiLuJe** [comment]
Yup, sounds sensible ;).

**Comment by poire-z** [comment]
Fixed by #7300 (commit taken from #7280 to be merged before).
